src/lib/hooks/swap/useConvenienceFee.ts

```diff
@@ -0,0 +1,43 @@
+import { Percent } from '@uniswap/sdk-core'
+import useActiveWeb3React from 'hooks/useActiveWeb3React'
+import { useAtom } from 'jotai'
+import { feeOptionsAtom } from 'lib/state/swap'
+import { useCallback, useEffect } from 'react'
+
+interface FeeOptionsArgs {
+  convenienceFee?: number
+  convenienceFeeRecipient?: string | string | { [chainId: number]: string }
+}
+
+export default function useConvenienceFee({ convenienceFee, convenienceFeeRecipient }: FeeOptionsArgs) {
```

Comment:
This is only ever used as a setter, so rename it (same with `useSwapDefaults`): `useSetConvenienceFee`; and don't return any value.